app/queries/common.py

```
@@ -291,6 +293,15 @@ def router_read_many[T: BaseModel, I: Identifiable](  # noqa: PLR0913
         .limit(pagination_request.page_size)
     )
 
+    # Add semantic similarity ordering if embedding is provided and model has embedding field
+    if embedding is not None and hasattr(db_model_class, "embedding"):
+        # Remove existing ordering clauses and replace with semantic similarity ordering
```

As quickly discussed before, this is an important thing to agree on, because it depends on the use case for the semantic search. It means that:
- the order of records is different (species, strain and brain_region are ordered by name by default. we don't allow ordering by other attributes, but it's technically possible)
- in general, the user cannot order the results by other attributes. In any case with this approach it could be better to explicitly raise an error if the order_by is explicitly passed by the user together with the semantic search, instead of silently ignore it.
- all the records are returned, even the ones with very low similarity (that will be at the bottom of the list or in the last pages)

---

> in general, the user cannot order the results by other attributes. In any case with this approach it could be better to explicitly raise an error if the order_by is explicitly passed by the user together with the semantic search, instead of silently ignore it.

We initially wanted to raise an exception but then we realized that somehow by default the swagger UI attaches the `order_by=name` query param **by default**. That is why we decided to just ignore it in case `semantic_search` is provided too. Alternatively, we could just somehow make sure the `order_by=name` is not sent by default. Feel free to decide.

> the order of records is different (species, strain and brain_region are ordered by name by default. we don't allow ordering by other attributes, but it's technically possible)

Point taken that one can provide multiple entries in the `order_by`. Now, there are two options
1. `semantic_search` comes first. Then I would argue there is no need for other tiebreaker entries (e.g. name, created_date,...) since no two entries in the DB will have the same distance. The current PR basically implements this case since it disregards all the other `order_by` clauses.
2. `semantic_search` **does not**  come first. If for some reason someone wants to sort by creation date first and only then by semantic search then I do agree that our current PR does not support it. However, I would argue that it is not that useful. However, it can be implemented but maybe I would leave that for a future PR since the iteraction between the FastAPI filter library and this custom `semantic_search` is not obvious.